pkg/config/state/state_manager_test.go

```diff
@@ -0,0 +1,286 @@
+	FloatField1  float32           `yaml:"float_field1,omitempty"`
+	FloatField2  float64           `yaml:"float_field2,omitempty"`
+	StringField1 string            `yaml:"string_field1,omitempty"`
+	StringField2 string            `yaml:"string_field2,omitempty"`
+	Sub          TestStateSubEntry `yaml:"sub,omitempty"`
+}
+
+type StateTestSuite struct {
+	suite.Suite
+}
+
+func (suite *StateTestSuite) TestGetReturnsCorrectState() {
+	var tests = []struct {
+		Key      types.StateEntryKey
+		Expected TestStateEntry
+	}{
+		// key1 to 3 are available in the test data.
+		{
+			Key: "key1",
+			Expected: TestStateEntry{
+				BoolField1:   true,
+				BoolField2:   false,
+				IntField1:    11,
+				IntField2:    12,
+				UIntField1:   112,
+				UIntField2:   123,
+				FloatField1:  1.1,
+				FloatField2:  1.2,
+				StringField1: "string11",
+				StringField2: "string12",
+			},
+		},
+		{
+			Key: "key2",
+			Expected: TestStateEntry{
+				BoolField1:   false,
+				BoolField2:   true,
+				IntField1:    21,
+				IntField2:    22,
+				UIntField1:   212,
+				UIntField2:   223,
+				FloatField1:  2.1,
+				FloatField2:  2.2,
+				StringField1: "string21",
+				StringField2: "string22",
+			},
+		},
+		{
+			Key: "key3",
+			Expected: TestStateEntry{
+				BoolField1:   true,
+				BoolField2:   true,
+				IntField1:    31,
+				IntField2:    32,
+				UIntField1:   312,
+				UIntField2:   323,
+				FloatField1:  3.1,
+				FloatField2:  3.2,
+				StringField1: "string31",
+				StringField2: "string32",
+			},
+		},
+		// Tests if the initialization works properly.
+		{
+			Key:      "dummy",
+			Expected: TestStateEntry{},
+		},
+	}
+
+	// Use the state file from the test data.
+	state := yaml.NewState(filepath.Join("testdata", "state.yml"))
+
+	// Variable must be initialized before the for loop to make the last test
+	// working which needs values to be set.
+	stateEntry := TestStateEntry{}
+
+	for _, tt := range tests {
+		suite.Run(tt.Key, func() {
+			suite.NoError(state.Get(tt.Key, &stateEntry))
+			suite.EqualValues(tt.Expected, stateEntry)
+		})
+	}
+}
+
+func (suite *StateTestSuite) TestSetUpdatesState() {
+	// Use a non existing state file from the temp dir.
+	state := yaml.NewState(filepath.Join(suite.T().TempDir(), "state.yml"))
+
+	stateEntryOut := TestStateEntry{
+		BoolField1:   true,
+		BoolField2:   false,
+		FloatField1:  12.34,
+		FloatField2:  23.45,
+		IntField1:    12,
+		IntField2:    23,
+		UIntField1:   123,
+		UIntField2:   234,
+		StringField1: "string1",
+		StringField2: "string2",
+		Sub: TestStateSubEntry{
+			BoolField1:   true,
+			IntField1:    1,
+			UIntField1:   2,
+			FloatField1:  3.1,
+			StringField1: "string1",
+		},
+	}
+	stateEntryIn := TestStateEntry{}
+
+	suite.NoError(state.Set("test_state_entry", stateEntryOut))
+	suite.NoError(state.Get("test_state_entry", &stateEntryIn))
+	suite.EqualValues(stateEntryOut, stateEntryIn)
+}
+
+func (suite *StateTestSuite) TestSetPreservesExistingStates() {
```

Comment:
Do these test suite items need to have global scope or should this be testSetPreservesExistingStates()?